### PR TITLE
feat: add get_auto_mine function [APE-998]

### PR DESCRIPTION
**Pull request by wakamex**
### What I did

add anvil's `get_autoMine` call so you can tell what kind of behavior your anvil done is up to. reference: <https://book.getfoundry.sh/reference/anvil/>

### How I did it

added this code:
```py
    def get_auto_mine(self) -> bool:
        return self._make_request("anvil_getAutoMine", [])
```

### How to verify it

install the plugin, connect to an anvil node, and run `provider.get_auto_mine()`.

using `anvil` default:
```py
print(f"{provider.get_auto_mine()=}")
provider.get_auto_mine()=True
```

with `anvil --block-time 1` that mines a new block every 1 second... automatically?
```py
print(f"{provider.get_auto_mine()=}")
provider.get_auto_mine()=False
```

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

**Review by antazoey — DISMISSED** [review]
optional: a test just for showcasing how it gets used.

File: ape_foundry/provider.py [hunk 1]
```diff
@@ -385,6 +385,9 @@ def set_balance(self, account: AddressType, amount: Union[int, float, str, bytes
     def set_timestamp(self, new_timestamp: int):
         self._make_request("evm_setNextBlockTimestamp", [new_timestamp])
 
+    def get_auto_mine(self) -> bool:
+        return self._make_request("anvil_getAutomine", [])
```
Comment (fubuloubu) [inline ape_foundry/provider.py:389]: just to be consistent with other read-only properties, this should probably be a property too
```suggestion
    @property
    def auto_mine(self) -> bool:
        return self._make_request("anvil_getAutomine", [])
```

Comment (antazoey) [reply ape_foundry/provider.py:389]: you gotta be careful with properties that makes requests under-the-hood:

```python
if self.auto_mine:
   # do things that cause self.auto_mine to change.
   cached_val = self.auto_mine # was expecting true but it changed to false
```

Comment (fubuloubu) [reply ape_foundry/provider.py:389]: fair, but also this is why we have the walrus operator :wink: 
```py
if cached_val := self.auto_mine:
   # do things that cause self.auto_mine to change.
```

Comment (antazoey) [reply ape_foundry/provider.py:389]: ya you can set it to a variable, that is all that does, that is like a workaround tho

Comment (antazoey) [reply ape_foundry/provider.py:389]: but actually... im ok with it being a property... i dont see it changing too often or anything

**Comment by wakamex** [comment]
I can add a test if y'all want, here's how I'm currently using it: 
```
automine = provider._make_request("anvil_getAutomine", parameters={})
if bot_config.devnet and automine:  # anvil automatically mines after you send a transaction. or manually.
    ape.chain.mine()
```

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**